components/dashboard/src/settings/Billing.tsx

```diff
@@ -10,11 +10,13 @@ import { BillingAccountSelector } from "../components/BillingAccountSelector";
 import { UserContext } from "../user-context";
 import UsageBasedBillingConfig from "../components/UsageBasedBillingConfig";
 import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import useStripe from "../hooks/useStripe";
 
 export default function Billing() {
     const { user } = useContext(UserContext);
-
     const attributionId: AttributionId = { kind: "user", userId: user?.id || "" };
+    const [showSpinner, _, showUpgradeBilling, showManageBilling, stripePortalUrl, usageLimit, doUpdateLimit] =
+        useStripe(user, attributionId, `pendingStripeSubscriptionForUser${user?.id || ""}`);
```

Comment:
I wonder if a more "idiomatic" solution to this problem could be a `StripeBillingContext` (similar to `UserContext`, `TeamContext`, `PaymentContext` etc).

Here are the changes that would be needed to achieve this:
- Refactor `useStripe.ts` to a `stripe-billing-context.tsx` (you can look at [theme-context.tsx](https://github.com/gitpod-io/gitpod/blob/0d757c4cd34111c46ffea402a3ed1c2a8d7e60c6/components/dashboard/src/theme-context.tsx) for inspiration on the structure)
- Import and use the `StripeBillingContextProvider` in [index.tsx](https://github.com/gitpod-io/gitpod/blob/6a1f7546c01e529cb35cddc88a3b7e739c09e415/components/dashboard/src/index.tsx)
- Use it in `Billing.tsx` (i.e. here) and `TeamUsageBasedBilling.tsx` like so:

```ts
const { showSpinner, showUpgradeBilling, showManageBilling, stripePortalUrl, usageLimit, doUpdateLimit } = useContext(StripeBillingContext);
```

Reply:
Ah, or, alternatively, it seems to me that you're getting a bunch of common variables out of one thing (`useStripe` hook or `StripeBillingContext`) but only to feed them unchanged and unused straight to another common component (`UsageBasedBillingConfig`).

Doesn't this mean that you could "simply" get all these common variables directly inside `UsageBasedBillingConfig`, and pass just the `subject` and `attributionId` to `UsageBasedBillingConfig`?